plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -423,7 +428,8 @@ public IcebergMetadata(
             Optional<HiveMetastoreFactory> metastoreFactory,
             boolean addFilesProcedureEnabled,
             Predicate<String> allowedExtraProperties,
-            ExecutorService executor)
+            ExecutorService executor,
+            Executor metadataFetchingExecutor)
```

Comment:
Now `executor` is too general. Please rename to `systemTableExecutor` or something.